runtime/src/tiered_storage/footer.rs

```diff
@@ -0,0 +1,331 @@
+use {
+    crate::tiered_storage::{file::TieredStorageFile, mmap_utils::get_type},
+    memmap2::Mmap,
+    serde::{Deserialize, Serialize},
+    solana_sdk::hash::Hash,
+    std::{mem, path::Path},
+};
+
+pub const FOOTER_FORMAT_VERSION: u64 = 1;
+
+// The size of the footer struct + the u64 magic number at the end.
+pub const FOOTER_SIZE: i64 = (mem::size_of::<TieredStorageFooter>() + mem::size_of::<u64>()) as i64;
+// The size of the ending part of the footer.  This size should remain unchanged
+// even when the footer's format changes.
+pub const FOOTER_TAIL_SIZE: i64 = 24;
+
+// The ending 8 bytes of a valid tiered account storage file.
+pub const FOOTER_MAGIC_NUMBER: u64 = 0x502A2AB5; // SOLALABS -> SOLANA LABS
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+#[repr(C)]
+pub struct TieredStorageMagicNumber(pub u64);
+
+impl Default for TieredStorageMagicNumber {
+    fn default() -> Self {
+        Self(FOOTER_MAGIC_NUMBER)
+    }
+}
+
+#[repr(u64)]
```

Comment:
This is to make all the fields in the footer naturally aligned.

Reply:
> Also, what does "naturally" aligned mean? Are the u32 fields getting padded out to align on 64-bit boundaries?

Yep, not sure why I was brain-dead about this.  It should be 8-bits alignment instead of 8-bytes xD.

Will reduce them to something smaller and have align(8).

Reply:
Look like align(8) isn't necessary here as well since everything should be already align(8).